controllers/operatorpolicy_controller.go

```diff
@@ -108,167 +101,300 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+		if err != nil {
+			return fmt.Errorf("error updating the status for a missing OperatorGroup: %w", err)
+		}
 
-	// Exists indicates if a Subscription or Operatorgroup need to be created
-	exists := subExists && ogExists
-	shouldExist := strings.EqualFold(string(policy.Spec.ComplianceType), string(policyv1.MustHave))
+		if policy.Spec.RemediationAction.IsEnforce() {
+			err = r.Create(ctx, desiredOpGroup)
+			if err != nil {
+				return fmt.Errorf("error creating the OperatorGroup: %w", err)
+			}
 
-	// Case 1: policy has just been applied and related resources have yet to be created
-	if !exists && shouldExist {
-		OpLog.Info("The object does not exist but should exist")
+			desiredOpGroup.SetGroupVersionKind(operatorGroupGVK) // Create stripped this information
 
-		return r.createPolicyResources(ctx, policy, cachedSubscription, cachedOperatorGroups)
-	}
+			// Now the OperatorGroup should match, so report Compliance
+			err = r.updateStatus(ctx, policy, createdCond("OperatorGroup"), createdObj(desiredOpGroup))
+			if err != nil {
+				return fmt.Errorf("error updating the status for a created OperatorGroup: %w", err)
+			}
+		}
+	case 1:
+		opGroup := foundOpGroups[0]
+
+		if policy.Spec.OperatorGroup != nil {
+			emptyNameMatch := desiredOpGroup.Name == "" && opGroup.GetGenerateName() == desiredOpGroup.GenerateName
+
+			if opGroup.GetName() == desiredOpGroup.Name || emptyNameMatch {
+				// check whether the specs match
+				desiredUnstruct, err := runtime.DefaultUnstructuredConverter.ToUnstructured(desiredOpGroup)
+				if err != nil {
+					return fmt.Errorf("error converting desired OperatorGroup to an Unstructured: %w", err)
+				}
+
+				if reflect.DeepEqual(desiredUnstruct["spec"], opGroup.Object["spec"]) {
+					// Everything relevant matches! This is a "happy path".
+					// Only update the condition if it reflects a NonCompliant state because emitting a
+					// Compliant -> Compliant 'change' is noisy and would hide *why* this part of the
+					// policy is currently compliant (eg that the policy previously updated the object).
+					idx, existingCond := policy.Status.GetCondition(opGroupConditionType)
+
+					if idx == -1 || existingCond.Status == metav1.ConditionFalse {
+						err := r.updateStatus(ctx, policy, matchesCond("OperatorGroup"), matchedObj(&opGroup))
+						if err != nil {
+							return fmt.Errorf("error updating the status for an OperatorGroup that matches: %w", err)
+						}
+					}
+				} else {
+					// The names match, but the specs don't: report NonCompliance
+					err := r.updateStatus(ctx, policy, mismatchCond("OperatorGroup"), mismatchedObj(&opGroup))
+					if err != nil {
+						return fmt.Errorf("error updating the status for an OperatorGroup that does not match: %w", err)
+					}
+
+					if policy.Spec.RemediationAction.IsEnforce() {
+						desiredOpGroup.ResourceVersion = opGroup.GetResourceVersion()
+
+						err := r.Update(ctx, desiredOpGroup)
```

Comment:
This seems like "must only have" since I don't see any checking to see if the desired operator group is a subset of the existing operator group.